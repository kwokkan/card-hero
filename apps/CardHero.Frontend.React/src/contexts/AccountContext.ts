import React from "react";
import { IUserModel } from "../clients/clients";

interface IAccountContextProps {
    user?: IUserModel;
    setUser: (user: IUserModel) => void;
}

export const AccountContext = React.createContext<IAccountContextProps>({ setUser: (_user: IUserModel) => { } });