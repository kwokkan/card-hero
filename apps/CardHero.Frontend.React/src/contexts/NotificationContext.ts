import React from "react";
import { INotificationItem } from "../types/INotificationItem";

export interface INotificationContextProps {
    notifications: INotificationItem[];
    addNotification: (notification: INotificationItem) => void;
    removeNotification: (notification: INotificationItem) => void;
}

export const NotificationContext = React.createContext<INotificationContextProps>({
    notifications: [],
    addNotification: (_notification: INotificationItem) => { },
    removeNotification: (_notification: INotificationItem) => { }
});
