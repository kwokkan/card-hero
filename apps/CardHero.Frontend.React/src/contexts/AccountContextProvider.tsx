import { ProviderProps, useContext, useState } from "react";
import { IUserModel } from "../clients/clients";
import { AccountContext, IAccountContextProps } from "./AccountContext";

export interface IAccountContextProviderProps {
    user?: IUserModel;
}

interface IAccountContextProviderState {
    user?: IUserModel;
}

function AccountContextProvider(props: ProviderProps<IAccountContextProviderProps>): JSX.Element {
    const [state, setState] = useState<IAccountContextProviderState>({
        user: props.value.user
    });

    const contextProps: IAccountContextProps = {
        setUser: (user) => {
            setState({
                user: user
            });
        },
        user: state.user
    };

    return (
        <AccountContext.Provider
            value={contextProps}
            children={props.children}
        />
    );
}

function useAccountContext() {
    return useContext(AccountContext);
}

function useAccountContextConsumer() {
    return AccountContext.Consumer;
}

export { AccountContextProvider, useAccountContext, useAccountContextConsumer };
