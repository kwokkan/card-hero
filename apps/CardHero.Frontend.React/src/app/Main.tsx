import React, { useEffect } from "react";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import { CardApp } from "../components/card/CardApp";
import { CardRoute } from "../components/card/CardRoute";
import { CollectionApp } from "../components/collection/CollectionApp";
import { DeckApp } from "../components/deck/DeckApp";
import { DeckRoute } from "../components/deck/DeckRoute";
import { GameApp } from "../components/game/GameApp";
import { GameRoute } from "../components/game/GameRoute";
import { HomeApp } from "../components/home/HomeApp";
import { NavMenu } from "../components/shared/NavMenu";
import { NotificationWidget } from "../components/shared/NotificationWidget";
import { StoreApp } from "../components/store/StoreApp";
import { useAccountContext } from "../contexts/AccountContextProvider";
import { AccountService } from "../services/AccountService";
import { getRoutePrefix } from "../utils/route";

interface IMainProps {
    appName: string;
    baseUrl?: string;
}

export function Main(props: IMainProps) {
    const { setUser, user: contextUser } = useAccountContext();

    useEffect(() => {
        async function loadData() {
            const user = await AccountService.getAccount();

            if (user) {
                setUser(user);
            }
        }
        loadData();
    }, []);  //eslint-disable-line react-hooks/exhaustive-deps

    const appName = props.appName;
    const baseUrl = getRoutePrefix(props.baseUrl);

    return (
        <BrowserRouter basename={baseUrl}>
            <NavMenu appName={appName} user={contextUser} />

            <div className="container-fluid body-content">
                <NotificationWidget />

                <div className="row">
                    <Routes>
                        <Route
                            path="/"
                            element={<HomeApp appName={appName} routePrefix={baseUrl} />}
                        />

                        <Route
                            path="/Card"
                            element={<CardApp routePrefix="/Card/" />}
                        >
                            <Route
                                path=":id"
                                element={<CardRoute />}
                            />
                        </Route>

                        <Route
                            path="/Game"
                            element={<GameApp routePrefix="/Game/" />}
                        >
                            <Route
                                path=":id"
                                element={<GameRoute />}
                            />
                        </Route>

                        <Route
                            path="/Store"
                            element={<StoreApp />}
                        />

                        <Route
                            path="/Collection"
                            element={<CollectionApp />}
                        />

                        <Route
                            path="/Deck"
                            element={<DeckApp routePrefix="/Deck/" />}
                        >
                            <Route
                                path=":id"
                                element={<DeckRoute />}
                            />
                        </Route>
                    </Routes>
                </div>
            </div>
        </BrowserRouter>
    );
}
