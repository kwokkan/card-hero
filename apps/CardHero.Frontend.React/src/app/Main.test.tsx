/*
 * @jest-environment jsdom
 */
import React from "react";
import renderer from "react-test-renderer";
import { AccountContextProvider, IAccountContextProviderProps } from "../contexts/AccountContextProvider";
import { Main } from "./Main";

test("<Main /> renders correctly without user", () => {
    const tree = renderer
        .create(
            <Main appName="Test app" />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<Main /> renders correctly with user", () => {
    const props: IAccountContextProviderProps = {
        user: {
            coins: 123456789,
            fullName: "Test user"
        }
    };
    const tree = renderer
        .create(
            <AccountContextProvider value={props}>
                <Main appName="Test app" />
            </AccountContextProvider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
