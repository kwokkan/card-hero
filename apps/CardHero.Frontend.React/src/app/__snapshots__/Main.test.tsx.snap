// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders correctly with user 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark navbar-expand-md fixed-top"
  >
    <button
      aria-controls="navbar-header"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbar-header"
      data-toggle="collapse"
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-fw "
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Test app"
        className="d-inline-block align-top"
        height="32"
        src="/favicon.ico"
        width="32"
      />
       
      Test app
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbar-header"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Card"
            onClick={[Function]}
          >
            Cards
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Game"
            onClick={[Function]}
          >
            Games
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Store"
            onClick={[Function]}
          >
            Store
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Collection"
            onClick={[Function]}
          >
            Collection
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Deck"
            onClick={[Function]}
          >
            Decks
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://github.com/kwokkan/card-hero"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-github fa-fw "
              data-icon="github"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            Fork me
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/swagger"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-code fa-fw "
              data-icon="code"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M414.8 40.79L286.8 488.8C281.9 505.8 264.2 515.6 247.2 510.8C230.2 505.9 220.4 488.2 225.2 471.2L353.2 23.21C358.1 6.216 375.8-3.624 392.8 1.232C409.8 6.087 419.6 23.8 414.8 40.79H414.8zM518.6 121.4L630.6 233.4C643.1 245.9 643.1 266.1 630.6 278.6L518.6 390.6C506.1 403.1 485.9 403.1 473.4 390.6C460.9 378.1 460.9 357.9 473.4 345.4L562.7 256L473.4 166.6C460.9 154.1 460.9 133.9 473.4 121.4C485.9 108.9 506.1 108.9 518.6 121.4V121.4zM166.6 166.6L77.25 256L166.6 345.4C179.1 357.9 179.1 378.1 166.6 390.6C154.1 403.1 133.9 403.1 121.4 390.6L9.372 278.6C-3.124 266.1-3.124 245.9 9.372 233.4L121.4 121.4C133.9 108.9 154.1 108.9 166.6 121.4C179.1 133.9 179.1 154.1 166.6 166.6V166.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            API
          </a>
        </li>
        <li
          className="navbar-text"
        >
          <span
            title="You have 123456789 coins"
          >
            <span
              className="coin-group"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-coins fa-fw coin-gold"
                data-icon="coins"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 80C512 98.01 497.7 114.6 473.6 128C444.5 144.1 401.2 155.5 351.3 158.9C347.7 157.2 343.9 155.5 340.1 153.9C300.6 137.4 248.2 128 192 128C183.7 128 175.6 128.2 167.5 128.6L166.4 128C142.3 114.6 128 98.01 128 80C128 35.82 213.1 0 320 0C426 0 512 35.82 512 80V80zM160.7 161.1C170.9 160.4 181.3 160 192 160C254.2 160 309.4 172.3 344.5 191.4C369.3 204.9 384 221.7 384 240C384 243.1 383.3 247.9 381.9 251.7C377.3 264.9 364.1 277 346.9 287.3C346.9 287.3 346.9 287.3 346.9 287.3C346.8 287.3 346.6 287.4 346.5 287.5L346.5 287.5C346.2 287.7 345.9 287.8 345.6 288C310.6 307.4 254.8 320 192 320C132.4 320 79.06 308.7 43.84 290.9C41.97 289.9 40.15 288.1 38.39 288C14.28 274.6 0 258 0 240C0 205.2 53.43 175.5 128 164.6C138.5 163 149.4 161.8 160.7 161.1L160.7 161.1zM391.9 186.6C420.2 182.2 446.1 175.2 468.1 166.1C484.4 159.3 499.5 150.9 512 140.6V176C512 195.3 495.5 213.1 468.2 226.9C453.5 234.3 435.8 240.5 415.8 245.3C415.9 243.6 416 241.8 416 240C416 218.1 405.4 200.1 391.9 186.6V186.6zM384 336C384 354 369.7 370.6 345.6 384C343.8 384.1 342 385.9 340.2 386.9C304.9 404.7 251.6 416 192 416C129.2 416 73.42 403.4 38.39 384C14.28 370.6 .0003 354 .0003 336V300.6C12.45 310.9 27.62 319.3 43.93 326.1C83.44 342.6 135.8 352 192 352C248.2 352 300.6 342.6 340.1 326.1C347.9 322.9 355.4 319.2 362.5 315.2C368.6 311.8 374.3 308 379.7 304C381.2 302.9 382.6 301.7 384 300.6L384 336zM416 278.1C434.1 273.1 452.5 268.6 468.1 262.1C484.4 255.3 499.5 246.9 512 236.6V272C512 282.5 507 293 497.1 302.9C480.8 319.2 452.1 332.6 415.8 341.3C415.9 339.6 416 337.8 416 336V278.1zM192 448C248.2 448 300.6 438.6 340.1 422.1C356.4 415.3 371.5 406.9 384 396.6V432C384 476.2 298 512 192 512C85.96 512 .0003 476.2 .0003 432V396.6C12.45 406.9 27.62 415.3 43.93 422.1C83.44 438.6 135.8 448 192 448z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
               
              12345
            </span>
            <span
              className="coin-group"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-coins fa-fw coin-silver"
                data-icon="coins"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 80C512 98.01 497.7 114.6 473.6 128C444.5 144.1 401.2 155.5 351.3 158.9C347.7 157.2 343.9 155.5 340.1 153.9C300.6 137.4 248.2 128 192 128C183.7 128 175.6 128.2 167.5 128.6L166.4 128C142.3 114.6 128 98.01 128 80C128 35.82 213.1 0 320 0C426 0 512 35.82 512 80V80zM160.7 161.1C170.9 160.4 181.3 160 192 160C254.2 160 309.4 172.3 344.5 191.4C369.3 204.9 384 221.7 384 240C384 243.1 383.3 247.9 381.9 251.7C377.3 264.9 364.1 277 346.9 287.3C346.9 287.3 346.9 287.3 346.9 287.3C346.8 287.3 346.6 287.4 346.5 287.5L346.5 287.5C346.2 287.7 345.9 287.8 345.6 288C310.6 307.4 254.8 320 192 320C132.4 320 79.06 308.7 43.84 290.9C41.97 289.9 40.15 288.1 38.39 288C14.28 274.6 0 258 0 240C0 205.2 53.43 175.5 128 164.6C138.5 163 149.4 161.8 160.7 161.1L160.7 161.1zM391.9 186.6C420.2 182.2 446.1 175.2 468.1 166.1C484.4 159.3 499.5 150.9 512 140.6V176C512 195.3 495.5 213.1 468.2 226.9C453.5 234.3 435.8 240.5 415.8 245.3C415.9 243.6 416 241.8 416 240C416 218.1 405.4 200.1 391.9 186.6V186.6zM384 336C384 354 369.7 370.6 345.6 384C343.8 384.1 342 385.9 340.2 386.9C304.9 404.7 251.6 416 192 416C129.2 416 73.42 403.4 38.39 384C14.28 370.6 .0003 354 .0003 336V300.6C12.45 310.9 27.62 319.3 43.93 326.1C83.44 342.6 135.8 352 192 352C248.2 352 300.6 342.6 340.1 326.1C347.9 322.9 355.4 319.2 362.5 315.2C368.6 311.8 374.3 308 379.7 304C381.2 302.9 382.6 301.7 384 300.6L384 336zM416 278.1C434.1 273.1 452.5 268.6 468.1 262.1C484.4 255.3 499.5 246.9 512 236.6V272C512 282.5 507 293 497.1 302.9C480.8 319.2 452.1 332.6 415.8 341.3C415.9 339.6 416 337.8 416 336V278.1zM192 448C248.2 448 300.6 438.6 340.1 422.1C356.4 415.3 371.5 406.9 384 396.6V432C384 476.2 298 512 192 512C85.96 512 .0003 476.2 .0003 432V396.6C12.45 406.9 27.62 415.3 43.93 422.1C83.44 438.6 135.8 448 192 448z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
               
              67
            </span>
            <span
              className="coin-group"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-coins fa-fw coin-bronze"
                data-icon="coins"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M512 80C512 98.01 497.7 114.6 473.6 128C444.5 144.1 401.2 155.5 351.3 158.9C347.7 157.2 343.9 155.5 340.1 153.9C300.6 137.4 248.2 128 192 128C183.7 128 175.6 128.2 167.5 128.6L166.4 128C142.3 114.6 128 98.01 128 80C128 35.82 213.1 0 320 0C426 0 512 35.82 512 80V80zM160.7 161.1C170.9 160.4 181.3 160 192 160C254.2 160 309.4 172.3 344.5 191.4C369.3 204.9 384 221.7 384 240C384 243.1 383.3 247.9 381.9 251.7C377.3 264.9 364.1 277 346.9 287.3C346.9 287.3 346.9 287.3 346.9 287.3C346.8 287.3 346.6 287.4 346.5 287.5L346.5 287.5C346.2 287.7 345.9 287.8 345.6 288C310.6 307.4 254.8 320 192 320C132.4 320 79.06 308.7 43.84 290.9C41.97 289.9 40.15 288.1 38.39 288C14.28 274.6 0 258 0 240C0 205.2 53.43 175.5 128 164.6C138.5 163 149.4 161.8 160.7 161.1L160.7 161.1zM391.9 186.6C420.2 182.2 446.1 175.2 468.1 166.1C484.4 159.3 499.5 150.9 512 140.6V176C512 195.3 495.5 213.1 468.2 226.9C453.5 234.3 435.8 240.5 415.8 245.3C415.9 243.6 416 241.8 416 240C416 218.1 405.4 200.1 391.9 186.6V186.6zM384 336C384 354 369.7 370.6 345.6 384C343.8 384.1 342 385.9 340.2 386.9C304.9 404.7 251.6 416 192 416C129.2 416 73.42 403.4 38.39 384C14.28 370.6 .0003 354 .0003 336V300.6C12.45 310.9 27.62 319.3 43.93 326.1C83.44 342.6 135.8 352 192 352C248.2 352 300.6 342.6 340.1 326.1C347.9 322.9 355.4 319.2 362.5 315.2C368.6 311.8 374.3 308 379.7 304C381.2 302.9 382.6 301.7 384 300.6L384 336zM416 278.1C434.1 273.1 452.5 268.6 468.1 262.1C484.4 255.3 499.5 246.9 512 236.6V272C512 282.5 507 293 497.1 302.9C480.8 319.2 452.1 332.6 415.8 341.3C415.9 339.6 416 337.8 416 336V278.1zM192 448C248.2 448 300.6 438.6 340.1 422.1C356.4 415.3 371.5 406.9 384 396.6V432C384 476.2 298 512 192 512C85.96 512 .0003 476.2 .0003 432V396.6C12.45 406.9 27.62 415.3 43.93 422.1C83.44 438.6 135.8 448 192 448z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
               
              89
            </span>
          </span>
        </li>
        <li
          className="nav-item dropdown"
        >
          <span
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link dropdown-toggle"
            data-toggle="dropdown"
            id="user-navbar-dropdown-menu"
          >
            Test user
          </span>
          <div
            aria-labelledby="user-navbar-dropdown-menu"
            className="dropdown-menu dropdown-menu-right"
          >
            <span
              className="dropdown-item"
            >
              Profile
            </span>
            <span
              className="dropdown-item"
            >
              Active Games
            </span>
            <div
              className="dropdown-divider"
            />
            <span
              className="dropdown-item"
            >
              Settings
            </span>
            <div
              className="dropdown-divider"
            />
            <a
              className="dropdown-item"
              href="/SignOut"
            >
              Logout
            </a>
          </div>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="container-fluid body-content"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <div
          className="jumbotron text-center"
        >
          <h1
            className="display-3"
          >
            Test app
          </h1>
          <p
            className="lead"
          >
            Test app
             is a dynamic action game.
          </p>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Cards
                </h4>
                <p
                  className="card-text"
                >
                  Discover over 1000 cards.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Card"
                  onClick={[Function]}
                >
                  View Cards
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Decks
                </h4>
                <p
                  className="card-text"
                >
                  Create you own decks.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Deck"
                  onClick={[Function]}
                >
                  View Decks
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Battle
                </h4>
                <p
                  className="card-text"
                >
                  Battle against other players.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Game"
                  onClick={[Function]}
                >
                  View Games
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`<Main /> renders correctly without user 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark navbar-expand-md fixed-top"
  >
    <button
      aria-controls="navbar-header"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbar-header"
      data-toggle="collapse"
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-fw "
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Test app"
        className="d-inline-block align-top"
        height="32"
        src="/favicon.ico"
        width="32"
      />
       
      Test app
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbar-header"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Card"
            onClick={[Function]}
          >
            Cards
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Game"
            onClick={[Function]}
          >
            Games
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Store"
            onClick={[Function]}
          >
            Store
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://github.com/kwokkan/card-hero"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-github fa-fw "
              data-icon="github"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            Fork me
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/swagger"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-code fa-fw "
              data-icon="code"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M414.8 40.79L286.8 488.8C281.9 505.8 264.2 515.6 247.2 510.8C230.2 505.9 220.4 488.2 225.2 471.2L353.2 23.21C358.1 6.216 375.8-3.624 392.8 1.232C409.8 6.087 419.6 23.8 414.8 40.79H414.8zM518.6 121.4L630.6 233.4C643.1 245.9 643.1 266.1 630.6 278.6L518.6 390.6C506.1 403.1 485.9 403.1 473.4 390.6C460.9 378.1 460.9 357.9 473.4 345.4L562.7 256L473.4 166.6C460.9 154.1 460.9 133.9 473.4 121.4C485.9 108.9 506.1 108.9 518.6 121.4V121.4zM166.6 166.6L77.25 256L166.6 345.4C179.1 357.9 179.1 378.1 166.6 390.6C154.1 403.1 133.9 403.1 121.4 390.6L9.372 278.6C-3.124 266.1-3.124 245.9 9.372 233.4L121.4 121.4C133.9 108.9 154.1 108.9 166.6 121.4C179.1 133.9 179.1 154.1 166.6 166.6V166.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            API
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/SignIn"
          >
            Sign in
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="container-fluid body-content"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <div
          className="jumbotron text-center"
        >
          <h1
            className="display-3"
          >
            Test app
          </h1>
          <p
            className="lead"
          >
            Test app
             is a dynamic action game.
          </p>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Cards
                </h4>
                <p
                  className="card-text"
                >
                  Discover over 1000 cards.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Card"
                  onClick={[Function]}
                >
                  View Cards
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Decks
                </h4>
                <p
                  className="card-text"
                >
                  Create you own decks.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Deck"
                  onClick={[Function]}
                >
                  View Decks
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Battle
                </h4>
                <p
                  className="card-text"
                >
                  Battle against other players.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Game"
                  onClick={[Function]}
                >
                  View Games
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
